workspace/src/app/views/shared/Metadata/MetadataUtils.tsx

```diff
@@ -48,6 +49,23 @@ export const queryTags = (projectId: string, filter?: string): Promise<FetchResp
         url: workspaceApi(`/projects/${projectId}/tags${filter?.length ? `?filter=${filter}` : ""}`),
     });
 
+export const updateMetaData = (
+    payload: Partial<{ label: string; description: string; tags: string[] }>,
+    projectId?: string,
+    taskId?: string
+): Promise<FetchResponse<IMetadata>> =>
+    projectId && taskId
+        ? fetch({
+              url: legacyApiEndpoint(`/projects/${projectId}/tasks/${taskId}/metadata`),
+              method: "put",
+              body: payload,
+          })
+        : fetch({
+              url: workspaceApi(`/projects/${projectId}/metaData`),
```

Comment:
When both `projectId` and `taskId` are undefined, this will go against `/projects/undefined/...`.